From a932ccc3163bff8771af4f522f03785bf3733415 Mon Sep 17 00:00:00 2001
From: Daniele Forsi <daniele@forsi.it>
Date: Sun, 31 Jul 2011 15:53:25 +0200
Subject: [PATCH] Fix "Buffer access out-of-bounds" found by cppcheck

Destination is 6 bytes while source is 6 byte + string terminator.
Use sizeof () on destination like a couple of lines below this.

Fixes:
[src/itdb_hash72.c:131]: (error) Buffer access out-of-bounds: hash_info.header
---
 src/itdb_hash72.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/itdb_hash72.c b/src/itdb_hash72.c
index e57c22f..ce61757 100644
--- a/src/itdb_hash72.c
+++ b/src/itdb_hash72.c
@@ -128,7 +128,7 @@ static gboolean write_hash_info (const Itdb_Device *device,
     gboolean success;
     const char header[] = "HASHv0";
 
-    memcpy (hash_info.header, header, sizeof (header));
+    memcpy (hash_info.header, header, sizeof (hash_info.header));
     success = itdb_device_get_hex_uuid (device, hash_info.uuid);
     if (!success) {
 	return FALSE;
-- 
1.7.4.1

